tests/tt_metal/tt_metal/unit_tests_fast_dispatch/command_queue/test_EnqueueProgram.cpp

```diff
@@ -1416,4 +1389,326 @@ TEST_F(CommandQueueFixture, TestRandomizedProgram) {
+        }
+        Program program = CreateProgram();
+        // Large eth kernels currently don't fit in the ring buffer, so we're reducing the max number of RTAs
+        // and the max kernel size to ensure that the kernel can fit in the ring buffer
+        this->create_kernel(
+            program,
+            CoreType::ETH,
+            false,
+            MIN_NUM_SEMS,
+            MAX_NUM_SEMS,
+            MIN_NUM_CBS,
+            MAX_NUM_CBS,
+            MIN_NUM_RUNTIME_ARGS,
+            MAX_NUM_RUNTIME_ARGS / 4,
+            MIN_KERNEL_SIZE_BYTES,
+            MAX_KERNEL_SIZE_BYTES / 2);
+        EnqueueProgram(device_->command_queue(), program, false);
+    }
+
+    Finish(device_->command_queue());
+}
+
+TEST_F(RandomProgramFixture, TestProgramsOnTensixAndEth) {
+    if (!local_test_functions::does_device_have_active_eth_cores(device_)) {
+        GTEST_SKIP() << "Skipping test because device " << device_->id() << "does not have any active ethernet cores";
+    }
+
+    for (uint32_t i = 0; i < NUM_PROGRAMS; i++) {
+        if (i % 10 == 0) {
+            log_info(tt::LogTest, "Creating Program {}", i);
+        }
+        Program program = CreateProgram();
+
+        bool eth_kernel_added_to_program = false;
+        if (rand() % 2 == 0) {
+            // Large eth kernels currently don't fit in the ring buffer, so we're reducing the max number of RTAs
+            // and the max kernel size to ensure that the kernel can fit in the ring buffer
+            this->create_kernel(
+                program,
+                CoreType::ETH,
+                false,
+                MIN_NUM_SEMS,
+                MAX_NUM_SEMS / 2,
+                MIN_NUM_CBS,
+                MAX_NUM_CBS,
+                MIN_NUM_RUNTIME_ARGS,
+                MAX_NUM_RUNTIME_ARGS / 4,
+                MIN_KERNEL_SIZE_BYTES,
+                MAX_KERNEL_SIZE_BYTES / 2);
+            eth_kernel_added_to_program = true;
+        }
+        if (rand() % 2 == 0 || !eth_kernel_added_to_program) {
+            this->create_kernel(program, CoreType::WORKER, false, MIN_NUM_SEMS, MAX_NUM_SEMS / 2);
+        }
+
+        EnqueueProgram(device_->command_queue(), program, false);
+    }
+
+    Finish(device_->command_queue());
+}
+
+TEST_F(RandomProgramFixture, TestAlternatingLargeAndSmallProgramsOnTensix) {
+    for (uint32_t i = 0; i < NUM_PROGRAMS; i++) {
+        if (i % 10 == 0) {
+            log_info(tt::LogTest, "Creating Program {}", i);
+        }
+        Program program = CreateProgram();
+
+        uint32_t min_num_sems;
+        uint32_t max_num_sems;
+        uint32_t min_num_cbs;
+        uint32_t max_num_cbs;
+        uint32_t min_num_rt_args;
+        uint32_t max_num_rt_args;
+        uint32_t min_size_bytes;
+        uint32_t max_size_bytes;
+        uint32_t min_runtime_microseconds;
+        uint32_t max_runtime_microseconds;
+        if (i % 2 == 0) {
+            min_num_sems = MAX_NUM_SEMS * (8.0 / 10);
```

Comment:
Seems to me you can put the creation of the Large and Small programs into a helper function. Will be easier to maintain, if you need to modify these tests in the future.